README.md

```diff
@@ -8,140 +8,130 @@
+* KeyboardSupportViewController - A sublcass of UIViewController which handles different keyboard support options.
+* KeyboardSupportConfiguration - A struct composed of different options which include:
+        * Auto scrollling when keyboard appears by passing in your UIScrollView
+        * Moving a view when keyboard appears by passing in a bottom constraint and constraint offest if necessary
+        * Easy dismissal of the keyboard by tapping outside a text field
+        * Use of the keyboard's "Return" key to navigate between text fields
+        * Use of your custom keyboard toolbar to navigate between text fields
+        * Callback when the "Done" button of your custom toolbar or keyboard "Return" button is tapped
+        
 ## Usage
-### KeyboardManager for using "Return" key.
-Create a KeyboardManager. Pass in your UITextFields. Make sure returnKeyNavigationEnabled is set to `true`. The order of the text fields determines the navigation order for traversing from one to the next.
+### KeyboardSupportViewController can be configured with different options.
+
+1. Supporting a UIScrollView or one of its subclasses.
 ``` swift
-class ViewController: UIViewController {
+import KeyboardSupport
+
+class ViewController: KeyboardSupportViewController {
+    @IBOutlet private var scrollView: UIScrollView!
     @IBOutlet private var textField1: UITextField!
     @IBOutlet private var textField2: UITextField!
-    private var keyboardManager: KeyboardManager?
+
 
     override func viewDidLoad() {
         super.viewDidLoad()
-
-        keyboardManager = KeyboardManager(textFields: [textField1, textField2], returnKeyNavigationEnabled: true)
+        
+        let textFields: [UITextField] = [textField1, textField2]
+        
+        let configuration = KeyboardSupportConfiguration(textFields: textFields, scrollView: scrollView, usesDismissalView: true, usesKeyboardNextButtons: true)
+        configureKeyboardSupport(with: configuration)
     }
 }
 ```
 
-You can also set the delegate of the KeyboardManager to get a callback when "Done" is tapped on the keyboard.
+2. Supporting a bottom constraint of a view.
+    Passing a constraint offset can be useful when dealing with tab bars.
 ``` swift
-override func viewDidLoad() {
-    super.viewDidLoad()
+import KeyboardSupport
 
-    keyboardManager = KeyboardManager(textFields: [textField1, textField2], configuresReturnKeys: true)
-    keyboardManager?.delegate = self
-}
-```
+class ViewController: KeyboardSupportViewController {
+    @IBOutlet private var bottomConstraint: NSLayoutConstraint!
 
-``` swift
-extension ViewController: KeyboardManagerDelegate {
-    func keyboardManagerDidTapDone(_ manager: KeyboardManager) {
-        // Handle "Done" tap
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        let configuration = KeyboardSupportConfiguration(bottomConstraint: bottomConstraint, constraintOffset: 49, usesDismissalView: true)
```

Comment:
Do you think we could update this readme example (and the code in the project) to use an actual tab bar height? I feel like hard-coding this is asking for trouble these days.

Reply:
I'll get the actual tab bar height and use it. Good suggestion.